### ISSUE TITLE
Failure during CFE_ES_ExitApp if app calls CFE_TBL_Unregister

### ISSUE THREAD

**Issue by pcooksey** [nasa/cFE#741]
**Describe the bug**
Should an app call CFE_TBL_Unregister?
On CFE_ES_ExitApp I get:
```
1980-012-14:03:21.25234 CFE_TBL:RemoveAccessLink-PutPoolBuf[0] Fail Stat=0xC4000007, Hndl=0x56709730, Buf=0x00000000
1980-012-14:03:21.25236 CFE_TBL:RemoveAccessLink-PutPoolBuf[0] Fail Stat=0xC4000007, Hndl=0x56709730, Buf=0x00000000
```

The [documentation](https://github.com/nasa/cFE/blob/master/docs/cFE%20Application%20Developers%20Guide.md#82-registering-tables) no longer says anything about it so I'm starting to think it shouldn't be used. The [sample_app](https://github.com/nasa/sample_app/blob/4a4b8e281802f244a2facdcb3c32238751c678f3/fsw/src/sample_app.c) in the cFS repo also doesn't call CFE_TBL_Unregister either.

The code seems to say we should though:
https://github.com/nasa/cFE/blob/95f34d25cb2843b8ba9db6338bb0b53cb3e38f92/fsw/cfe-core/src/inc/cfe_tbl.h#L334-L339
 
**To Reproduce**
1. Register table
2. Unregister table
3. Call CFE_ES_ExitApp

**Expected behavior**
No errors on exit

**Code snips**
N/A

**System observed on:**
 - Hardware: Laptop
 - OS: Linux
 - Versions: Versions:cFE 6.7.7.0, OSAL 5.0.6.0, PSP 1.4.4.0

**Additional context**
N/A

**Reporter Info**
Philip Cooksey, NASA Ames

**Comment by pcooksey** [comment]
It seems by default the tables are being cleaned up automatically as shown below which is part of the CFE_ES_CleanUpApp function:
https://github.com/nasa/cFE/blob/d217ca39f1de74695f0d4e0eb427f90fbe8b1156/fsw/cfe-core/src/es/cfe_es_apps.c#L1083-L1085

This clean up call eventually calls:
https://github.com/nasa/cFE/blob/95f34d25cb2843b8ba9db6338bb0b53cb3e38f92/fsw/cfe-core/src/tbl/cfe_tbl_internal.c#L1429

Which is the same function in CFE_TBL_Unregister:
https://github.com/nasa/cFE/blob/d217ca39f1de74695f0d4e0eb427f90fbe8b1156/fsw/cfe-core/src/tbl/cfe_tbl_api.c#L641

Seems like a bug since an already unregistered table shouldn't be cleaned up again by CFE_ES_CleanUpApp -> CFE_TBL_CleanUpApp.

**Comment by skliper** [comment]
> Seems like a bug since an already unregistered table shouldn't be cleaned up again by CFE_ES_CleanUpApp -> CFE_TBL_CleanUpApp.

Agreed.  Updated title and marked as bug.

**Comment by jphickey** [comment]
Tested this today - It may have been fixed by other changes, possibly the transition to ES background task and the better locking ES that was implemented with that.

To test - what I did was modify SAMPLE_APP (which registers a table) to only run for 5 seconds, then exit.  As part of exiting, I added a call to `CFE_TBL_Unregister()` before it calls `CFE_ES_ExitApp()` ... I confirmed that the first call to CFE_TBL_Unregister is indeed doing the un-registration, followed by ES invoking `CFE_TBL_CleanUpApp()` via the background task - which did nothing in my test.

The only (minor) issue I observed is that there is an extra "unknown state" event after the app exits itself:
`EVS Port1 66/1/CFE_ES 48: ES_ProcControlReq: Unknown State Application SAMPLE_APP ( 1 )`

But otherwise everything seemed to work just fine.
